src/chrome/chromeDebugAdapter.ts

```diff
@@ -646,75 +655,86 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
-        this.doAfterProcessingSourceEvents(async () => { // This will block future 'removed' source events, until this processing has been completed
-            if (typeof this._columnBreakpointsEnabled === 'undefined') {
-                await this.detectColumnBreakpointSupport(script.scriptId).then(async () => {
+        const breakpointsAreResolvedDefer = this.getBreakpointsResolvedDefer(script.scriptId);
+        try {
+            this.doAfterProcessingSourceEvents(async () => { // This will block future 'removed' source events, until this processing has been completed
+                if (typeof this._columnBreakpointsEnabled === 'undefined') {
+                    await this.detectColumnBreakpointSupport(script.scriptId);
                     await this.sendInitializedEvent();
-                });
-            }
+                }
+
+                if (this._earlyScripts) {
+                    this._earlyScripts.push(script);
+                } else {
+                    await this.sendLoadedSourceEvent(script);
+                }
+            });
 
-            if (this._earlyScripts) {
-                this._earlyScripts.push(script);
+            if (script.url) {
+                script.url = utils.fixDriveLetter(script.url);
             } else {
-                await this.sendLoadedSourceEvent(script);
+                script.url = ChromeDebugAdapter.EVAL_NAME_PREFIX + script.scriptId;
             }
-        });
 
-        if (script.url) {
-            script.url = utils.fixDriveLetter(script.url);
-        } else {
-            script.url = ChromeDebugAdapter.EVAL_NAME_PREFIX + script.scriptId;
-        }
+            this._scriptsById.set(script.scriptId, script);
+            this._scriptsByUrl.set(this.fixPathCasing(script.url), script);
 
-        this._scriptsById.set(script.scriptId, script);
-        this._scriptsByUrl.set(this.fixPathCasing(script.url), script);
+            const mappedUrl = await this._pathTransformer.scriptParsed(script.url);
 
-        const resolvePendingBPs = (source: string) => {
-            source = source && this.fixPathCasing(source);
-            const pendingBP = this._pendingBreakpointsByUrl.get(source);
-            if (pendingBP && !pendingBP.bpsSet) {
-                this.resolvePendingBreakpoint(pendingBP)
-                    .then(() => this._pendingBreakpointsByUrl.delete(source));
-            }
-        };
+            const resolvePendingBPs = async (source: string) => {
+                source = source && this.fixPathCasing(source);
+                const pendingBP = this._pendingBreakpointsByUrl.get(source);
+                if (pendingBP && !pendingBP.bpsSet) {
+                    await this.resolvePendingBreakpoint(pendingBP);
+                    this._pendingBreakpointsByUrl.delete(source);
+                    breakpointsAreResolvedDefer.resolve();
+                } else {
+                    breakpointsAreResolvedDefer.resolve();
```

Comment:
You can just pull these two out of the if/else

Reply:
I didn't noticed that, thanks :)... It's actually a bug to do these here, so I'm removing both of them....